woocommerce-services.php

```diff
@@ -667,6 +668,95 @@ public function localize_and_enqueue_service_script( $method_id, $instance_id =
 		}
 
 		/**
+		 * Add tracking info (if available) to completed emails using the woocommerce_email_after_order_table hook
+		 *
+		 * @param $order
+		 * @param $sent_to_admin
+		 * @param $plain_text
+		 */
+		public function add_tracking_info_to_emails( $order, $sent_to_admin, $plain_text ) {
+			$id = WC_Connect_Compatibility::instance()->get_order_id( $order );
+
+			// Abort if no id was passed or if the order is not marked as 'completed'
+			if ( ! $id || ! $order->has_status( 'completed' )) {
+				return;
+			}
+
+			$labels = $this->service_settings_store->get_label_order_meta_data( $id );
+
+			// Abort if there are no labels
+			if ( empty( $labels ) ) {
+				return;
+			}
+
+			$markup = '';
+
+			// Generate a table row for each label
+			foreach ( $labels as $label ) {
+				$carrier = $label['carrier_id'];
+				$carrierLabel = strtoupper( $carrier );
+				$tracking = $label['tracking'];
+				$refunded = array_key_exists( 'refund', $label );
+
+				if ( $refunded ) {
+					continue;
+				}
+
+				if ( $plain_text ) {
+					// Should look like '- USPS: 9405536897846173912345' in plain text mode
+					$markup .= '- ' . $carrierLabel . ': ' . $tracking . "\n";
+					continue;
+				}
+
+				$markup .= '<tr>';
+				$markup .= '<td class="td" scope="col">' . esc_html( $carrierLabel ) . '</td>';
+
+				switch ( $carrier ) {
+					case 'fedex':
+						$tracking_url = 'https://www.fedex.com/apps/fedextrack/?action=track&tracknumbers=' . $tracking;
+						break;
+					case 'usps':
+						$tracking_url = 'https://tools.usps.com/go/TrackConfirmAction.action?tLabels=' . $tracking;
+						break;
+				}
+
+				$markup .= '<td class="td" scope="col">';
+				$markup .= '<a href="' . esc_url( $tracking_url ) . '">' . esc_html( $tracking ) . '</a>';
```

Comment:
In my testing, this link ended up being white against a white background. The culprit was the colors selected in my email settings and the interplay with `wc_light_or_dark()` used to choose a color for header text and also links in emails: https://github.com/woocommerce/woocommerce/blob/master/templates/emails/email-styles.php#L188

Let's force the tracking link text to use the color in the `woocommerce_email_text_color` option.